migration-engine/connectors/sql-migration-connector/src/sql_destructive_change_checker/destructive_change_checker_flavour/mysql.rs

```diff
@@ -129,27 +129,48 @@ impl DestructiveChangeCheckerFlavour for MysqlFlavour {
     }
 
     fn count_rows_in_table<'a>(&'a mut self, table: &'a Table) -> BoxFuture<'a, ConnectorResult<i64>> {
+        // TODO(MultiSchema): replace this when implementing MySQL.
+        let query = format!("SELECT COUNT(*) FROM `{}`", table.table);
+
         Box::pin(async move {
-            // TODO(MultiSchema): replace this when implementing MySQL.
-            let query = format!("SELECT COUNT(*) FROM `{}`", table.table);
-            let result_set = self.query_raw(&query, &[]).await?;
-            super::extract_table_rows_count(table, result_set)
+            query_with_backoff(self, &query)
+                .await
+                .and_then(|result_set| super::extract_table_rows_count(table, result_set))
         })
     }
 
     fn count_values_in_column<'a>(&'a mut self, column: &'a Column) -> BoxFuture<'a, ConnectorResult<i64>> {
+        // TODO(MultiSchema): replace this when implementing MySQL.
+        let query = format!(
+            "SELECT COUNT(*) FROM `{}` WHERE `{}` IS NOT NULL",
+            column.table, column.column
+        );
+
         Box::pin(async move {
-            // TODO(MultiSchema): replace this when implementing MySQL.
-            let query = format!(
-                "SELECT COUNT(*) FROM `{}` WHERE `{}` IS NOT NULL",
-                column.table, column.column
-            );
-            let result_set = self.query_raw(&query, &[]).await?;
-            super::extract_column_values_count(result_set)
+            query_with_backoff(self, &query)
+                .await
+                .and_then(super::extract_column_values_count)
         })
     }
 }
 
+/// Run the query with exponential backoff on error, from 400ms up to (400 × 2⁵)ms.
```

Comment:
will push a commit expanding on _why_ we need retries here (tl;dr schema changes have become async in vitess)